std/experimental/allocator/building_blocks/free_list.d

```diff
@@ -1045,6 +1047,28 @@ struct SharedFreeList(ParentAllocator,
         resetNodes();
         return result;
     }
+
+    /**
+    Nonstandard function that minimizes the memory usage of the freelist by
+    freeing each element in turn. Defined only if $(D ParentAllocator) defines
+    $(D deallocate).
+    */
+    static if (hasMember!(ParentAllocator, "deallocate") && !unchecked)
+    void minimize() shared
+    {
+        lock.lock();
+        scope(exit) lock.unlock();
```

Comment:
Yes, I agree.  Check out SharedFreeList.deallocateAll(); it has the same concern, and possibly even longer lock-holding time because the parent allocator could do even more deallocation.  I'm not sure I really considered this when I first looked at it.  I'm pretty sure I used deallocateAll() as a model for my implementation; use the wisdom of others and whatnot.

Maybe the lock is necessary for deallocateAll() because deallocateAll is part of the uniform interface for allocators.  Minimize() does not have such a constraint.

Should SharedFreeList and FreeList instead (or additionally) have a function for returning memory to the parent allocator one node at a time instead of all at once?  If that were true, the lock time would be reduced inside SharedFreeList, and the caller could use whatever locking or synchronization mechanism they want if they intend to hold a lock continuously for all deallocations.  I suppose I just don't feel comfortable making that decision for the caller unless there is a clear Right Answer.